ikarus/python/finiteelements/linearelastic.hh

```diff
@@ -44,6 +44,20 @@ namespace Ikarus::Python {
 template <class LinearElastic, class... options>
 void registerLinearElastic(pybind11::handle scope, pybind11::class_<LinearElastic, options...> cls) {
   registerElement(scope, cls);
+
+  using Traits         = typename LinearElastic::Traits;
+  using FERequirements = typename LinearElastic::FERequirementType;
+
+  cls.def(
+      "calculateAt",
+      [](LinearElastic& self, const FERequirements& req, const Dune::FieldVector<double, Traits::mydim>& local,
+         std::string resType) {
+        if (resType == "linearStress")
+          return self.template calculateAt<ResultType::linearStress>(req, local).asVec();
```

Comment:
This should be a function `registerCalculateAt`, which accepts the result types and creates the registration. Then all the element register the function by calling

```cpp
registerCalculateAt(scope,cls,std::make_tuple(linearStress,....));
```

Reply:
You can again iterate over the tuple with Hybrid:forEach and if the string matches one toString(RT), then you can return, the corresponding value. Note that since calculateAt is called with the string the result type is only know at runtime, therefore in the end all the result have to be converted to `Eigen::VectorXd` or directly to `py::array_t<double>`.

Reply:
Okay yeah ah function is good for that, but how then I need a list somewhere of all `RT`s right? 
And i cant iterate over the namespace ... We don't have reflection :D

Reply:
For now you can pass that list by hand for each element as in `registerCalculateAt(scope,cls,std::make_tuple(linearStress,....));` but the elements can also export exaclty this tuple that they support.

Reply:
But i still have to match the templates to the strings somehow?

Reply:
I think thats what i try to indicate above
First we may use https://gitlab.dune-project.org/core/dune-common/-/blob/master/dune/common/tuplevector.hh instead of std::tuple since it has the convenient operator[]
```cpp
template<Element>
void registerCalculateAt(scope,cls,tuple)
{
  using Traits         = typename Element::Traits;
  using FERequirements = typename Element::FERequirementType;
  cls.def(
      "calculateAt",
      [](Element& self, const FERequirements& req, const Dune::FieldVector<double, Traits::mydim>& local,
         std::string resType) {
        Dune::Hypbrid::forEach(tuple,[&](auto i){
        if (resType == toString(tuple[i]))
          return Eigen::VectorXd(self.template calculateAt<tuple[i]>(req, local).asVec());
});
          DUNE_THROW(Dune::NotImplemented, "The requested result type is not supported by ");
      },
      pybind11::arg("feRequirements"), pybind11::arg("local"), pybind11::arg("resultType"));


});
}

Reply:
You can even create a string of supported types which can be used in the error message:
std::string restypes;
Dune::Hypbrid::forEach(tuple,[&](auto i){
restypes = restypes + toString(tuple[i]) + ",";
});

Reply:
Okay it works now, but it got a bit ugly, as i was not able to return out of the lamba. 
Please have a look